File: back/internal/order_event/domain/order_event.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// OrderEvent 订单事件
type OrderEvent struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	EventID string `gorm:"size:50;uniqueIndex;not null" json:"eventId"`
	OrderCode string `gorm:"size:50;not null;index" json:"orderCode"`
	EventType string `gorm:"size:50;not null" json:"eventType"`
	EventContent string `gorm:"type:text" json:"eventContent"`
	Operator string `gorm:"size:100;not null" json:"operator"`
	OperatedAt time.Time `json:"operatedAt"`
	CreatedAt time.Time      `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt time.Time      `gorm:"autoUpdateTime" json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}

// TableName 表名
func (OrderEvent) TableName() string {
	return "order_events"
}

File: front/apps/web-ele/src/api/core/order_event.ts
import { requestClient } from '../request';

export interface OrderEvent {
  id: number;
  eventId: string;
  orderCode: string;
  eventType: string;
  eventContent: string;
  operator: string;
  operatedAt: string;
  createdAt: string;
  updatedAt: string;
}

export interface OrderEventListResponse {
  total: number;
  list: OrderEvent[];
}

export async function createOrderEvent(data: {
  eventId: string;
  orderCode: string;
  eventType: string;
  eventContent?: string;
  operator: string;
  operatedAt: string;
}) {
  const response = await requestClient.post<OrderEvent>('/order_events', data);
  return response;
}

export async function getOrderEvents(params?: { limit?: number; offset?: number }) {
  const response = await requestClient.get<OrderEventListResponse>('/order_events', {
    params: {
      limit: params?.limit || 100,
      offset: params?.offset || 0,
    },
  });
  return response;
}

export async function getOrderEvent(id: number) {
  const response = await requestClient.get<OrderEvent>(`/order_events/${id}`);
  return response;
}

export async function updateOrderEvent(
  id: number,
  data: {
    eventId?: string;
    orderCode?: string;
    eventType?: string;
    eventContent?: string;
    operator?: string;
    operatedAt?: string;
  }
) {
  const response = await requestClient.post<OrderEvent>(`/order_events/${id}`, data);
  return response;
}

export async function deleteOrderEvent(id: number) {
  const response = await requestClient.delete(`/order_events/${id}`);
  return response;
}
